package/MDAnalysis/transformations/translate.py

```diff
@@ -138,4 +140,170 @@ def wrapped(ts):
+    
+    Parameters
+    ----------
+    ag: AtomGroup
+        atom group to be centered on the unit cell.
+    axis: str
+        used to define the plane on which the given AtomGroup will be centered. Defined as
+        a string or an array-like of the plane and the value of its translation. Suported 
+        planes are x, y and z.
+    origin: array-like or str
+        coordinate on which the axes are centered. Can be an array of three coordinates or
+        `center` to shift the origin to the center of the unit cell. Default is `None` 
+        which sets [0, 0, 0] as the origin of the axis. 
+    center: str, optional
+        used to choose the method of centering on the given atom group. Can be 'geometry'
+        or 'mass'
+    wrap: bool, optional
+        If `True`, all the atoms from the given AtomGroup will be moved to the unit cell
+        before calculating the center of mass or geometry. Default is `False`, no changes
+        to the atom coordinates are done before calculating the center of the AtomGroup. 
+    
+    Returns
+    -------
+    :class:`~MDAnalysis.coordinates.base.Timestep` object
+    
+    """
+    pbc_arg = wrap
+    if isinstance(axis, string_types):
+        if axis not in ('x', 'y', 'z'):
+            raise ValueError('{} is not a valid axis'.format(axis))
+    if origin is not None:
+        if isinstance(origin, string_types):
+            if not origin == 'center':
+                raise ValueError('{}is not a valid origin'.format(origin))
+        else:
+            origin = np.asarray(origin, np.float32)
+            if origin.shape != (3, ) and origin.shape != (1, 3):
+                raise ValueError('{} is not a valid origin'.format(origin))
+    else:
+        origin = np.asarray([0, 0, 0], np.float32)
+    try:
+        if center == 'geometry':
+            center_method = partial(ag.center_of_geometry, pbc=pbc_arg)
+        elif center == 'mass':
+            center_method = partial(ag.center_of_mass, pbc=pbc_arg)
+        else:
+            raise ValueError('{} is not a valid argument for center'.format(center))
+    except AttributeError:
+        if center == 'mass':
+            raise AttributeError('{} is not an AtomGroup object with masses'.format(ag))
+        else:
+            raise ValueError('{} is not an AtomGroup object'.format(ag))
+    
+    def wrapped(ts):
+        if isinstance(origin, string_types) and origin == 'center':
+            _origin = ts.triclinic_dimensions.sum(axis=0) / 2.0
+        else:
+            _origin = origin
+        ag_center = center_method()
+        if axis == 'x':
+            center = np.asarray([ag_center[0], _origin[1], _origin[2]], np.float32)
+        if axis == 'y':
+            center = np.asarray([_origin[0], ag_center[1], _origin[2]], np.float32)
+        if axis == 'z':
+            center = np.asarray([_origin[0], _origin[1], ag_center[2]], np.float32)
+        vector = center - ag_center
```

Comment:
`center_of_mass` and `center_of_geometry` output an array of dtype float32. I can typecast it again, but it isn't really necessary

Reply:
There is no need to typecast again. Though, it would be good to have a test for each transformation to make sure the coordinates have the correct dtype.